### PR TITLE
feat: (testing) add temporal dev server testing utility for e2e tests

### PR DESCRIPTION
**Pull request by laouji**
relates to ENG-1510

### CODE REVIEW

File: testing/platform/temporaltesting/temporal.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"context"
+	"io"
+
+	"github.com/stretchr/testify/require"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/testsuite"
+)
+
+var defaultNamespace = "temporaltesting-default-namespace"
```
Comment (Dav-14) [inline testing/platform/temporaltesting/temporal.go:12]: not configurable ?

File: testing/platform/temporaltesting/temporal.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+type TemporalT interface {
+	require.TestingT
+	Cleanup(func())
```
Comment (Dav-14) [inline testing/platform/temporaltesting/temporal.go:16]: Cleanup is not used, how does it stop ?

**Review by Dav-14 — APPROVED** [review]

File: testing/platform/temporaltesting/temporal.go (hunk 2 above)
Comment (laouji) [reply testing/platform/temporaltesting/temporal.go:16]: Good point. I'll add it.

File: testing/platform/temporaltesting/temporal.go (hunk 1 above)
Comment (laouji) [reply testing/platform/temporaltesting/temporal.go:12]: At the moment we don't have any applications using more than one namespace, so it didn't seem super important to add this feature.

File: testing/platform/temporaltesting/temporal.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	}
+	t.Cleanup(func() {
+		_ = srv.Stop()
+	})
+	return &TemporalServer{DevServer: srv, defaultNamespace: defaultNamespace}
+}
+
+func (s *TemporalServer) Client(ctx context.Context) client.Client {
+	return s.Client(ctx)
+}
```
Comment (Dav-14) [inline testing/platform/temporaltesting/temporal.go:41]: Can we have more than one client ? for // testing

Comment (laouji) [reply testing/platform/temporaltesting/temporal.go:41]: testsuite.DevServer pre-emptively creates a client configured to communicate with the DevServer and this method gives access to that particular preconfigured client. But there's nothing stopping the tester from using the `DefaultNamespace` and `HostPort` methods to configure additonal clients using any of the constructors provided by the go-sdk client package.

**Pull request merged**